packages/settings-library/src/settings_library/node_ports.py

```diff
@@ -0,0 +1,56 @@
+import json
+from datetime import timedelta
+from typing import Any, Final
+
+from pydantic import Field, NonNegativeInt, PositiveInt, SecretStr, root_validator
+from pydantic.json import pydantic_encoder
+
+from .base import BaseCustomSettings
+from .postgres import PostgresSettings
+from .storage import StorageSettings
+
+NODE_PORTS_400_REQUEST_TIMEOUT_ATTEMPTS_DEFAULT_VALUE: Final[NonNegativeInt] = 3
+
+
+class StorageAuthSettings(StorageSettings):
+    STORAGE_USERNAME: str | None
+    STORAGE_PASSWORD: SecretStr | None
+
+    @property
+    def were_credentials_provided(self) -> bool:
+        return self.STORAGE_USERNAME is not None and self.STORAGE_PASSWORD is not None
+
+    @classmethod
+    @root_validator
+    def both_fields_set_or_unset(cls, values):
+        username = values["STORAGE_USERNAME"]
+        password = values["STORAGE_PASSWORD"]
+        if (username is None) != (password is None):
+            msg = f"Both {username=} and {password=} must be either set or unset!"
+            raise ValueError(msg)
+        return values
+
+    def unsafe_dict(self):
```

Comment:
this naming looks really scarry...
is the jsonable_encoder does not already handle secret strings??
@pcrespov?

Reply:
This was made on purpose so that it would not expose credentials when printing the settings.
I required a json in clear when passing this in the env var encoded as json for the dy-sidecar.

I have dropped this in favour of single env vars. Will these 2 functions